titus_isolate/model/opportunistic_resource.py

```diff
@@ -0,0 +1,186 @@
+        endpoint the client submits requests to. Cannot be updated. In CamelCase.
+        More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#types-kinds
+        :param kind: The kind of this OpportunisticResource.
+        :type: str
+        """
+
+        self._kind = kind
+
+    @property
+    def metadata(self):
+        """
+        Gets the metadata of this OpportunisticResource.
+        :return: The metadata of this OpportunisticResource.
+        :rtype: V1ObjectMeta
+        """
+        return self._metadata
+
+    @metadata.setter
+    def metadata(self, metadata):
+        """
+        Sets the metadata of this OpportunisticResource.
+        :param metadata: The metadata of this OpportunisticResource.
+        :type: V1ObjectMeta
+        """
+
+        self._metadata = metadata
+
+    @property
+    def spec(self):
+        """
+        Gets the spec of this OpportunisticResource.
+        Spec describes how the user wants the resources to appear
+        :return: The spec of this OpportunisticResource.
+        :rtype: OpportunisticResourceSpec
+        """
+        return self._spec
+
+    @spec.setter
+    def spec(self, spec):
+        """
+        Sets the spec of this OpportunisticResource.
+        Spec describes how the user wants the resources to appear
+        :param spec: The spec of this OpportunisticResource.
+        :type: OpportunisticResourceSpec
+        """
+        if spec is None:
+            raise ValueError("Invalid value for `spec`, must not be `None`")
+
+        self._spec = spec
+
+    def to_dict(self):
+        """
+        Returns the model properties as a dict
+        """
+        result = {}
+
+        for attr, _ in iteritems(self.swagger_types):
+            value = getattr(self, attr)
+            if isinstance(value, list):
+                result[attr] = list(map(
+                    lambda x: x.to_dict() if hasattr(x, "to_dict") else x,
+                    value
+                ))
+            elif hasattr(value, "to_dict"):
+                result[attr] = value.to_dict()
+            elif isinstance(value, dict):
+                result[attr] = dict(map(
+                    lambda item: (item[0], item[1].to_dict())
+                    if hasattr(item[1], "to_dict") else item,
+                    value.items()
+                ))
+            else:
+                result[attr] = value
+
+        return result
+
+    def to_str(self):
+        """
+        Returns the string representation of the model
+        """
+        return pformat(self.to_dict())
+
+    def __repr__(self):
+        """
+        For `print` and `pprint`
+        """
+        return self.to_str()
+
+    def __eq__(self, other):
+        """
+        Returns true if both objects are equal
+        """
+        if not isinstance(other, OpportunisticResource):
+            return False
+
+        return self.__dict__ == other.__dict__
+
+    def __ne__(self, other):
```

Comment:
Is this still needed in `Python 3.5+`?

Reply:
probably not. these classes are all copypasta from kubernetes

Reply:
should I delete them or leave? I'm no python guru so I'm not exactly sure which are required and which are not.

Reply:
Leave it.  Doesn't hurt.